src/main/java/io/soos/integration/domain/manifest/Manifest.java

```diff
@@ -100,30 +98,45 @@ public long sendManifests(String projectId, String analysisId, List<File> direct
 
             if (packageManagerPaths.size() > 0) {
                 this.LOG.info("Files: {}", packageManagerPaths.stream().map(path -> path.getFileName().toString()).collect(Collectors.toList()));
-                allPaths.addAll(packageManagerPaths);
+                allPaths.merge(packageManager, packageManagerPaths, (oldValue, newValue) -> {
+                    oldValue.addAll(newValue);
+                    return oldValue;
+                });
             } else {
                 this.LOG.info("No files found.");
             }
         });
 
         this.LOG.info("--------------------------------------------------------");
-
-        List<Path> pathsToUpload = allPaths;
-
-        try {
-            boolean hasMoreThanMaximumManifests = pathsToUpload.size() > Constants.MAX_MANIFESTS;
-            if (hasMoreThanMaximumManifests) {
-                pathsToUpload = pathsToUpload.subList(0, Constants.MAX_MANIFESTS);
-                this.LOG.info("Maximum number of manifests exceeded. Taking first {} only.", Constants.MAX_MANIFESTS);
+        int totalManifests = 0;
+        for (Map.Entry<String, List<Path>> entry : allPaths.entrySet()) {
+            if(totalManifests == Constants.MAX_MANIFESTS){
+                this.LOG.info("Maximum number of manifests reached ({}). Skipping remaining manifests.", Constants.MAX_MANIFESTS);
+                break;
             }
-            if (pathsToUpload.size() > 0) {
-                this.exec(projectId, analysisId, pathsToUpload, hasMoreThanMaximumManifests);
+            String packageManager = entry.getKey();
+            List<Path> pathsToUpload = entry.getValue();
+            totalManifests += pathsToUpload.size();
+            try {
+                boolean hasMoreThanMaximumManifests = totalManifests > Constants.MAX_MANIFESTS;
+                if (hasMoreThanMaximumManifests) {
+                    pathsToUpload = pathsToUpload.subList(0, Constants.MAX_MANIFESTS - (totalManifests - pathsToUpload.size()));
+                    this.LOG.info("Maximum number of manifests exceeded ({}). Taking first {} only.", Constants.MAX_MANIFESTS, pathsToUpload.size());
```

Comment:
"Taking first x only." is going to show the number of manifests in this batch for x which I think is going to confuse a client since the number being uploaded per package manager isn't really relevant to them. I'd just use the same message as above:

```suggestion
                    this.LOG.info("Maximum number of manifests reached ({}). Skipping remaining manifests.", Constants.MAX_MANIFESTS)
```